Api/JourneyHub.Api/Services/TripServices.cs

```diff
@@ -26,5 +34,45 @@ public async Task<Trip> CreateTripAsync(PostTripRequestDto tripDto)
 
             return trip;
         }
+
+        public async Task<string> getAreaByCoordinates(getAreaByCoordsDto areadto)
+        {
+            string lon = areadto.lon.ToString();
+            string lat = areadto.lat.ToString();
+
+            string _address = "https://nominatim.openstreetmap.org/reverse?lat="+ lat + "&lon="+ lon + "&format=json";
+
+            var client = new HttpClient();
+
+            var productValue = new ProductInfoHeaderValue("ScraperBot", "1.0");
+            var commentValue = new ProductInfoHeaderValue("(+http://www.example.com/ScraperBot.html)");
+
+            client.DefaultRequestHeaders.UserAgent.Add(productValue);
+            client.DefaultRequestHeaders.UserAgent.Add(commentValue);
+
+            client.BaseAddress = new Uri(_address);
+            HttpResponseMessage response = await client.GetAsync(new Uri(_address));
+            response.EnsureSuccessStatusCode();
+            var result = await response.Content.ReadAsStringAsync();
```

Comment:
If possible this could be separate service that could handle multiple different requests from parameters like URL, generic types of request and return. For example 
` ApiResponse<CalculationResultWrapper> responseWithWrapper = await _client.PostAsync<CalculationResultWrapper, CalculationParams>(_urlOptions.PACalculator, calculationParams, apiKeyType: "PA", cancellationToken);`

Reply:
More code provided in DC dm.